Add article Feature Flags and How to Iterate Quickly (#143)
open-feature/docs.openfeature.dev: blog/10-10-2022-feature-flags-and-how-to-iterate-quickly.md

---
slug: feature-flag-how-to-iterate-quickly
title: Feature Flags and How to Iterate Quickly
authors: [thomaspoignant]
tags: [blog]
unlisted: false
---

# Feature Flags and How to Iterate Quickly
## Why your first PR for a large feature should be its feature flag

> *Feature flags* (also known as *feature toggles* or *feature flipping*) is a technique in development that allows you to enable or disable a feature without deploying any codes.

Reviewer: ```suggestion
*Feature flagging* (also known as *feature toggles* or *feature flipping*) is a technique used in development to enable or disable features without deploying any codes.
```

The logic behind this is to deploy a conditional feature you can activate only for certain groups of users at a time. If the flag is `on`, the code is executed; if `off`, the code is skipped.

> This is a good practice in dev ops that allows you to test your feature directly in production.

Reviewer: ```suggestion
This is a good practice in DevOps because it allows you to test your feature directly in production without impacting customers.
```

## How It Works

This is really simple: You have a central configuration file or system which stores your flag.

> When you deploy a new feature in your code, you add an interrogation to this central configuration to know if this feature should be allowed for this specific user. If yes, the flag is `on` and your user can see the feature. If `off`, the user uses the application without the new feature.

Reviewer: ```suggestion
When you deploy a new feature in your code, you verify with the central configuration to know if this feature should be allowed for a specific user. If yes, the flag is `on` and your user can see the feature. If `off`, the user doesn't see the new feature.
```

I'm a bit unclear what you meant by the first sentence. Please let me know if I misunderstand what you were trying to convey.

There are a ton of frameworks and products to help you do this.

> One thing you should never forget is to remove your flag when you want to enable your feature for everybody.

Reviewer: I'm not sure if this sentence should be in the "How It Works" section. Perhaps it could go in the conclusion or in a new section that covers good practices.
